### PR TITLE
Update extensions to Keycloak 20.0.1

### PR DESCRIPTION
**Pull request by jens-hammel**
Update extensions and docker-comopse-file to use keycloak version 20.0.1.
Remove explicitly declared test-dependency of resteasy-jackson2-provider to avoid conflicting dependencies.
Improve wording in messages_de.properties. Message match test from postman-collection.

### CODE REVIEW

File: remove-user-on-logout/src/test/java/de/intension/listener/RemoveUserOnLogOutEventIT.java [hunk 1]
```diff
@@ -205,7 +210,7 @@ private KeycloakPage updateLastName(String lastName)
 
         public KeycloakPage logout()
         {
-            String signOutButton = "Sign Out";
+            String signOutButton = "Sign out";
```
Comment (ingokuba) [inline remove-user-on-logout/src/test/java/de/intension/listener/RemoveUserOnLogOutEventIT.java:213]: Did the label of the button change?

File: whitelist-authenticator/src/main/resources/theme-resources/messages/messages_de.properties [hunk 2]
```diff
@@ -1 +1 @@
-idpNotConfigured=Diese Anwendung an dieser Anwendung ist mit Deinem Nutzerkonto derzeit nicht m\u00F6glich. Bei Fragen wende dich bitte an den Support deines Portals oder an den VIDIS Support <a href="https://vidis.schule/support">vidis.schule/support</a>
+idpNotConfigured=Die Anmeldung an dieser Anwendung ist mit Deinem Nutzerkonto derzeit nicht m\u00F6glich. Bei Fragen wende dich bitte an den Support deines Portals oder an den VIDIS Support <a href="https://vidis.schule/support">vidis.schule/support</a>
```
Comment (ingokuba) [inline whitelist-authenticator/src/main/resources/theme-resources/messages/messages_de.properties:1]: Sneaky find :wink:

File: .gitignore [hunk 3]
```diff
@@ -1 +1,55 @@
...
+### JetBrains template
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# IntelliJ
+.idea/
+out/
+# File-based project format
+*.iws
+*.iml
```
Comment (ingokuba) [inline .gitignore:55]: Not a fan of adding IDE specific files to project gitignores https://blog.steven266.de/global-gitignore-files-2087e64da812
Will add a comment to CONTRIBUTING.md if that's ok

**Review by ingokuba — CHANGES_REQUESTED** [review]
Please update the version to `3.0.0` in pom.xml as described [here](https://github.com/FWU-DE/fwu-kc-extensions/blob/main/CONTRIBUTING.md#automatic-release-process)

File: .gitignore (hunk 3 above)
Comment (jens-hammel) [reply .gitignore]: Removed

**Pull request merged**